proposed/2022/dotnet-nuget-why-proposal.md

```diff
@@ -0,0 +1,139 @@
+
+- PACKAGE_NAME
+
+The package for whom the dependency graph has to be identified. Package name wildcards are not supported.
+
+#### Options
+
+- -f|--framework <FRAMEWORK>
+
+Print out the dependency graph of a given package for a specific target framework.
+
+- -?|-h|--help
+
+Prints out a description of how to use the command.
+
+#### Examples
+
+- List dependency graph of a package given `package id`.
+
+```
+dotnet nuget why packageA
+
+Project 'projectNameA' has the following dependency graph for 'packageA'
+   [net6.0]: 
+      Microsoft.ML (1.0.0) -> Microsoft.ML.Util (1.0.0) -> packageA (1.0.0)
+   [net472]:
+      Microsoft.ML (1.0.0) -> Microsoft.ML.Util (1.0.0) -> packageA (1.0.0)
+
+Project 'projectNameB' has the following dependency graph for 'packageA'
+   [net6.0]:
+      Microsoft.ML (1.1.0) -> Microsoft.ML.Util (1.1.0) -> packageA (1.1.0)
+   [net472]:
+      Microsoft.ML (1.1.0) -> Microsoft.ML.Util (1.1.0) -> packageA (1.1.0)
+```
+
+- List dependency graph of a package given `package id` when there is a diamond dependency (a package is brought in by more than one path).
+
+```
+dotnet nuget why packageA
+
+Project 'projectNameA' has the following dependency graph for 'packageA'
+   [net6.0]:
+      Microsoft.ML (1.0.0) -> Microsoft.ML.Util (1.0.0) -> packageA (1.0.0)
+      Microsoft.ML (1.0.0) -> Microsoft.ML.SampleUtils (1.0.0) -> packageA (1.0.0)
+```
+
+- List dependency graph of a package given `package id` and `target framework`.
+
+```
+dotnet nuget why packageA -f net6.0
+
+Project 'projectNameA' has the following dependency graph for 'packageA'
+   [net6.0]:
+      Microsoft.ML (1.0.0) -> Microsoft.ML.Util (1.0.0) -> packageA (1.0.0)
+
+Project 'projectNameB' has the following dependency graph for 'packageA'
+   [net6.0]:
+      Microsoft.ML (1.1.0) -> Microsoft.ML.Util (1.1.0) -> packageA (1.1.0)
+```
+
+## Limitations 
+
+The NuGet restore operation downloads other packages during dependency resolution which are not part of the final dependency graph that the `dotnet nuget why` command prints out. The absence of a package id and version in the `project.assets.json` file indicates that a package was downloaded during dependency resolution but is not part of the final dependency graph.
+
+## Rationale and Alternatives
+
+We could have considered a minor tweak to the existing experience of “dotnet list package --include-transitive” to provide the user with a sense of a package's dependencies. This could have been a new column next to “Resolved” called “Transitively Referenced” that had a list of the top-level packages that requested the dependency. The corresponding tracking issue can be found here: https://github.com/NuGet/Home/issues/11625. Unfortunately, this solution doesn't provide an easy way to identify the dependency graph `(starting from project -> top-level package -> transitive-dependency (1) -> ..-> transitive dependency (n))` for a particular package.
```

Comment:
I think I put this in the original transitive dependencies spec as a CLI enhancement. I'm not sure it will scale well when there's more than a couple of transitively referenced parents. I think a separate command is the right approach here. Good callout!

Reply:
It was @kartheekp-ms 's idea to include this, so thanks Kartheek!